backend/src-common/src/main/java/org/eclipse/sw360/datahandler/db/ComponentDatabaseHandler.java

```diff
@@ -609,42 +619,53 @@ public RequestStatus updateRelease(Release release, User user, Iterable<Release.
 
-            if (!hasChangesInEccFields && hasEmptyEccFields(release)) {
-                autosetEccFieldsForReleaseWithDownloadUrl(release);
-            }
+                if (!hasChangesInEccFields && hasEmptyEccFields(release)) {
+                    autosetEccFieldsForReleaseWithDownloadUrl(release);
+                }
 
-            copyFields(actual, release, immutableFields);
+                copyFields(actual, release, immutableFields);
 
-            autosetReleaseClearingState(release, actual);
-            if (hasChangesInEccFields) {
-                autosetEccUpdaterInfo(release, user);
-            }
-            release.setAttachments( getAllAttachmentsToKeep(toSource(actual), actual.getAttachments(), release.getAttachments()) );
-            deleteAttachmentUsagesOfUnlinkedReleases(release, actual);
-            releaseRepository.update(release);
-            updateReleaseDependentFieldsForComponentId(release.getComponentId());
-            //clean up attachments in database
-            attachmentConnector.deleteAttachmentDifference(nullToEmptySet(actual.getAttachments()),nullToEmptySet(release.getAttachments()));
-            sendMailNotificationsForReleaseUpdate(release, user.getEmail());
-        } else {
-            if (hasChangesInEccFields) {
-                return releaseModerator.updateReleaseEccInfo(release, user);
+                autosetReleaseClearingState(release, actual);
+                if (hasChangesInEccFields) {
+                    autosetEccUpdaterInfo(release, user);
+                }
+                release.setAttachments( getAllAttachmentsToKeep(toSource(actual), actual.getAttachments(), release.getAttachments()) );
+                deleteAttachmentUsagesOfUnlinkedReleases(release, actual);
+                releaseRepository.update(release);
+                updateReleaseDependentFieldsForComponentId(release.getComponentId());
+                //clean up attachments in database
+                attachmentConnector.deleteAttachmentDifference(nullToEmptySet(actual.getAttachments()),nullToEmptySet(release.getAttachments()));
+                sendMailNotificationsForReleaseUpdate(release, user.getEmail());
             } else {
-                return releaseModerator.updateRelease(release, user);
+                if (hasChangesInEccFields) {
+                    return releaseModerator.updateReleaseEccInfo(release, user);
+                } else {
+                    return releaseModerator.updateRelease(release, user);
+                }
             }
-        }
 
         return RequestStatus.SUCCESS;
+       }
     }
 
-    private void deleteAttachmentUsagesOfUnlinkedReleases(Release updated, Release actual) throws SW360Exception {
+    private boolean duplicateAttachmentExist(Release release) {
+    	if(release.attachments != null && !release.attachments.isEmpty()) {
+            return RepositoryUtils.isDuplicateAttachment(release.attachments);
+        }
+		return false;
```

Comment:
Please check spaces vs. tabs.

Reply:
Thanks @imaykay  for reviewing the changes.  I have added logic to check the sha1 aloong with file name check. 
As discussed about the second point we will restrict users from updating the object if there is a duplicate attachment present on the list and application will display an error message which will indicate the reason for  update failure. So I am not making any change for this as the logic is already added.